tools/dkcm/clerk/clerk.go

```diff
@@ -0,0 +1,112 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package clerk
+
+import (
+	"database/sql"
+	"fmt"
+	"time"
+
+	"knative.dev/test-infra/pkg/mysql"
+)
+
+// Cluster stores a row in the "Cluster" db table
+// Table Schema: knative.dev/test-infra/tools/dkcm/clerk/schema.sql
+type Cluster struct {
+	ClusterId   string
+	accessToken string
+	BoskosId    string
+	ProwId      string
+	Status      string
+	Zone        string
+	lastUpdate  time.Time
+}
+
+// ClusterConfig is the struct to return to Prow once the cluster is available
+type ClusterConfig struct {
+	ClusterName string
+	BoskosId    string
+	Zone        string
+}
+
+type ClerkOperations interface {
+	//check cluster available, if available, return cluster id and access token
+	CheckAvail() (bool, string, string)
+	// get with token
+	GetCluster(token string) (ClusterConfig, error)
+	// delete a cluster entry
+	DeleteCluster(accessToken string) error
+	// Insert a cluster entry
+	InsertCluster(accessToken string, boskosId string, prowId string, status string, zone string, lastUpdate time.Time) error
+	// List clutsers within a time interval
+	ListClusters(window time.Duration) ([]Cluster, error)
+	// Generate Unique access token for Prow
+	GenerateToken() string
```

Comment:
I think it makes sense to me, but should this be a public function? The flow in my mind is:

1. Request comes in
2. mainservice sends request to clerk
3. clerk returns a token right away

This function should only be used by clerk I think